pandas/io/_util.py

```diff
@@ -8,6 +8,9 @@
 def _arrow_dtype_mapping() -> dict:
     pa = import_optional_dependency("pyarrow")
     return {
+        # All nulls should still give Float64 not object
+        # TODO: This breaks parquet
+        # pa.null(): pd.Float64Dtype(),
```

Comment:
In what way does it break parquet?

Reply:
Well, first I hit this error.
```
>>> import pandas as pd
>>> import pyarrow as pa
>>> a = pa.nulls(0)
>>> pd.Float64Dtype().__from__arrow__(a)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: 'Float64Dtype' object has no attribute '__from__arrow__'
```
After patching that(another PR incoming),
now an empty DataFrame of dtype object(``pd.DataFrame({"value": pd.array([], dtype=object)})``) returns a Float64Dtype when roundtripped.

Best guess here is that somehow the ``types_mapper`` is overriding the pandas metadata somehow.